### PR TITLE
Avoid groups initialization  if required params aren't defined

### PR DESCRIPTION
**Pull request by ityaptin**
In ceilometer collector we don't need a initialization of decoder,
encoder, input and output if amqp, influxdb or elasticsearch connections
are not defined.

### CODE REVIEW

**Comment by cloudlab-autobuild** [comment]
Can one of the admins verify this patch?

**Comment by ityaptin** [comment]
@elemoine, @simonpasquier, @swann, @akomarek: Little fix for ceilometer_collector related configs

**Review by elemoine — CHANGES_REQUESTED** [review]
@ityaptin would you mind doing the same for the other services (log_collector, etc.)? You can do that with additional commits.

**Review by SwannCroiset — APPROVED** [review]

**Review by elemoine — APPROVED** [review]

**Review by elemoine — CHANGES_REQUESTED** [review]
@dkalashnik reported that this breaks the "heka" state.

**Review by dkalashnik — CHANGES_REQUESTED** [review]
Failing state.sls for heka

File: heka/meta/heka.yml [hunk 1]
```diff
@@ -459,10 +468,12 @@ aggregator:
       max_buffer_size: 1048576
       max_file_size: 524288
       full_action: drop
+  {%- endif %}
 {%- endif %}
+{%- if ceilometer_collector.influxdb_host is defined or ceilometer_collector.elasticsearch_host is defined or ceilometer_collector.amqp_host is defined %}
```
Comment (elemoine) [inline heka/meta/heka.yml]: Please remove this.

@ityaptin and I discussed this on Slack, and we decided to make `heka_alarming.py:grains_for_mine` more robust.

**Review by elemoine — CHANGES_REQUESTED** [review]

**Review by elemoine — APPROVED** [review]
Thanks for finding and fixing the bug in `heka_alarming.py`.

**Pull request merged**